### PR TITLE
Added verify_puzzle function on lib.rs

### PR DESCRIPTION
**Pull request by yshwanth**
Updated the example code of actix server

### CODE REVIEW

**Comment by i5hi** [comment]
few nits to address:

rust doesnt require you to use ; when returning . If you dont use ; it implies return, and also doesnt need the return key word. 

you only need to use the return keyword if you want to return from inside an inner scope, for example from inside a match statement.

```rust
 if check < error_margin {
        return true;
    } else {
        return false;
    }
  ```

can just be

```rust    
     if check < error_margin {
      true
    } else {
        false
    }
    ```

**Comment by i5hi** [comment]
there are some parts of the example where you have added ; to some returns, these are also unecessary

**Comment by i5hi** [comment]
also noticed you have made some formatting changes, so we might as well unify it across the project. 

1. use cargo clippy
2. get rust-analyzer plugin in vscode which auto formats when you use Ctrl+Shift+i.
3. Make sure tabs are configured to 4 spaces
4. Where possible group imports in {}

**Comment by yshwanth** [comment]
```let check = (solution - submission).abs();```
 ```   check < error_margin```

This is valid too, running cargo clippy

**Comment by i5hi** [comment]
> `let check = (solution - submission).abs();` ` check < error_margin`
> 
> This is valid too, running cargo clippy

you can go one step better and avoid creating a variable check - will be more memory optimized

File: examples/actix.rs [hunk 1]
```diff
@@ -80,7 +79,7 @@ async fn verify_handler(state: Data<State>, solution: web::Json<Solution>) -> im
                 "SOLUTION:\nRequestID:{:?}\nx:{:?}\n",
                 solution.id, correct_solution
             );
-            correct_solution.clone()
+            *correct_solution
```
Comment (i5hi) [inline examples/actix.rs:74]: nice one!

**Comment by i5hi** [comment]
Good one!

Thanks!

**Pull request merged**